clients/general/client.go

```diff
@@ -30,6 +29,8 @@ type GeneralClient interface {
 	FetchConfiguration(ctx context.Context) (string, error)
 	// FetchMetrics obtains metrics information from the target service.
 	FetchMetrics(ctx context.Context) (string, error)
+	// SetConfiguration sets configuration information into the target service.
+	SetConfiguration(service string, key string, value string, ctx context.Context) error
```

Comment:
Why doesn't this take the currently named `SetConfig` struct instead of inline `key / value` params? Since we're setting configuration for a service, can that also be added as a member to the `SetConfig` type? What sort of validation does this input require?

Reply:
- That is so because I need to conform to [the API exposed to the external client](https://github.com/akramtexas/edgex-go/blob/54c1b839ac20f617170c5463d90bbaed6dbfcd83/internal/system/agent/router.go#L37-L36) (e.g. `Pulse IoT`, etc), leading up to the [configHandler](https://github.com/akramtexas/edgex-go/blob/54c1b839ac20f617170c5463d90bbaed6dbfcd83/internal/system/agent/router.go#L162) itself.
- Making a note of your feedback `What sort of validation does this input require?` 👍